hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -79,32 +83,44 @@ public ContainerHealthTask(
   }
 
   @Override
-  public synchronized void run() {
+  public void run() {
     try {
+      lock.writeLock().lock();
```

Comment:
Since writelock is taken before  while loop, then external trigger will always wait for the writelock, deadlock situation.

Reply:
Thanks @sumitagrawl for review, its fixed. pls re-review.